homeassistant/components/device_automation/condition.py

```diff
@@ -55,19 +56,40 @@ async def async_get_conditions(
 class DeviceCondition(Condition):
     """Device condition."""
 
-    def __init__(self, hass: HomeAssistant, config: ConfigType) -> None:
-        """Initialize condition."""
-        self._config = config
-        self._hass = hass
+    _hass: HomeAssistant
+    _config: ConfigType
+
+    @classmethod
+    async def async_validate_complete_config(
+        cls, hass: HomeAssistant, complete_config: ConfigType
+    ) -> ConfigType:
+        """Validate complete config."""
+        complete_config = await async_validate_device_automation_config(
+            hass,
+            complete_config,
+            cv.DEVICE_CONDITION_SCHEMA,
+            DeviceAutomationType.CONDITION,
+        )
+        # Since we don't want to migrate device conditions to a new format
+        # we just pass the entire config as options.
+        complete_config[CONF_OPTIONS] = complete_config.copy()
+        return complete_config
 
     @classmethod
     async def async_validate_config(
         cls, hass: HomeAssistant, config: ConfigType
     ) -> ConfigType:
-        """Validate device condition config."""
-        return await async_validate_device_automation_config(
-            hass, config, cv.DEVICE_CONDITION_SCHEMA, DeviceAutomationType.CONDITION
-        )
+        """Validate config.
+
+        This is here just to satisfy the abstract class interface. It is never called.
+        """
+        return config
```

Comment:
```suggestion
        raise NotImplementedError
```
if it should not be called, lets raise instead to ensure that it is not called.